compiler/rustc_middle/src/lint.rs

```diff
@@ -384,10 +381,22 @@ pub fn struct_lint_level<'s, 'd>(
                 "this borrowing pattern was not meant to be accepted, and may become a hard error \
                  in the future"
                     .to_owned()
-            } else if let Some(edition) = future_incompatible.edition {
-                format!("{} in the {} edition!", STANDARD_MESSAGE, edition)
+            } else if let FutureIncompatibilityReason::EditionError(edition) =
+                future_incompatible.reason
+            {
+                let current_edition = sess.edition();
+                format!(
+                    "this is valid in the current edition (Rust {}) but is not accepted in the Rust {} edition!",
```

Comment:
In all other diagnostics we just say `"Rust 2021"` instead of things like `"the Rust 2021 edition"`.

What's the reason you changed the wording from "accepted / hard error" to "valid / not accepted"? I think the first one might be clearer.